### PR TITLE
Bring RKE2 docs up to date with new CIS 1.23

### PR DESCRIPTION
**Pull request by dereknola**
Copy over work from https://github.com/rancher/rke2/pull/3360 into the new docs site, with minor changes to match the "docusaurus" style guides.

Signed-off-by: Derek Nola <derek.nola@suse.com>

### CODE REVIEW

File: docs/security/hardening_guide.md [hunk 1]
```diff
@@ -2,104 +2,264 @@
...
+
 This section gives you the commands necessary to configure your host to meet the above requirements.
 
 #### Set kernel parameters
-When RKE2 is installed, it creates a sysctl config file to set the required parameters appropriately.
-However, it does not automatically configure the Host to use this configuration. You must do this manually.
-The location of the config file depends on the installation method used.
 
-If RKE2 was installed via RPM, YUM, or DNF (the default on OSes that use RPMs, such as CentOS), run the following command(s):
+When RKE2 is installed, it creates a sysctl config file to set the required parameters appropriately. However, it does not automatically configures the host to use this configuration. You must do this manually. The location of the config file depends on the installation method used.
```
Comment (VestigeJ) [inline docs/security/hardening_guide.md:81]: Remove s on configures

"it does not automatically configure" vs configures

File: docs/security/cis_self_assessment123.md [hunk 2]
```diff
@@ -2276,25 +2252,26 @@ If you need to run containers which require hostPID, this should be defined in a
...
 <details>
 <summary>Rationale</summary>
-
 A container running in the host's IPC namespace can use IPC to interact with processes outside the container.
 
-There should be at least one PodSecurityPolicy (PSP) defined which does not permit containers to share the host IPC namespace.
+There should be at least one admission control policy defined which does not permit containers to share the host IPC namespace.
 
-If you have a requirement to containers which require hostIPC, this should be defined in a separate PSP and you should carefully check RBAC controls to ensure that only limited service accounts and users are given permission to access that PSP.
+If you need to run containers which require hostIPC, this should be definited in a separate policy and you should carefully check to ensure that only limited service accounts and users are given permission to use that policy.
```
Comment (VestigeJ) [inline docs/security/cis_self_assessment123.md:2274]: ** " this should be defined " not definited

**Review by VestigeJ — CHANGES_REQUESTED** [review]
Just the two spelling changes and then it looks good to me

File: docs/security/cis_self_assessment123.md (hunk 2 above)
Comment (dereknola) [reply docs/security/cis_self_assessment123.md:2274]: LOL, this language is pulled directly from the CIS offical benchmark https://paper.bobylive.com/Security/CIS/CIS-Kubernetes-Benchmark-V1_23--PDF.pdf see section 5.2.4. 

I will fix their misspelling :)

**Review by briandowns — APPROVED** [review]

**Pull request merged**